common/djangoapps/third_party_auth/pipeline.py

```diff
@@ -997,22 +998,28 @@ def get_username(strategy, details, backend, user=None, *args, **kwargs):  # lin
         else:
             username = uuid4().hex
 
-        short_username = (username[:max_length - uuid_length]
-                          if max_length is not None
-                          else username)
+        input_username = username
         final_username = slug_func(clean_func(username[:max_length]))
 
         # Generate a unique username for current user using username
         # as base but adding a unique hash at the end. Original
         # username is cut to avoid any field max_length.
-        # The final_username may be empty and will skip the loop.
         # We are using our own version of user_exists to avoid possible case sensitivity issues.
-        while not final_username or len(final_username) < min_length or user_exists({'username': final_username}):
+        # We will attempt different length random suffixes
+        while len(final_username) < min_length or user_exists({'username': final_username}):
+            this_uuid_length = randint(1, uuid_length)
```

Comment:
what if, instead of using 1 as the min of this random int, you use like `max(uuid_length / 2, 1)`, so that you don't randomly end up with suffix lengths of 1 so frequently?